mu/interface/editor.py

```diff
@@ -86,6 +90,44 @@ def __init__(self, path, text, newline=NEWLINE):
         self.breakpoint_lines = set()
         self.configure()
 
+    def dropEvent(self, event):
+        """
+        Run by Qt when *something* is dropped on this editor
+        """
+
+        # Does the drag event have any urls?
+        # Files are transfered as a url (by path not value)
+        if event.mimeData().hasUrls():
+            # Qt doesn't seem to have an 'open' action,
+            # this seems the most appropriate
+            event.setDropAction(Qt.CopyAction)
+            # Valid links
+            links = []
+            # Iterate over each of the urls attached to the event
+            for url in event.mimeData().urls():
+                # Check the url is to a local file
+                # (not a webpage for example)
+                if url.isLocalFile():
+                    # Grab a 'real' path from the url
+                    path = url.toLocalFile()
+                    # If the dropped file is a script or hex
+                    if os.path.splitext(path)[1].lower() in ['.py', '.hex']:
```

Comment:
Is there any disadvantage in letting all files pass through, and then have the `Editor._load` decide which files to process? This way there is also an informative message box saying if the file is not compatible.

Reply:
+1 to re-using existing functionality for this.

Reply:
OK... I'll fix this to use existing functionality.